open-hackathon-adminUI/src/app/admin/login.py

```diff
@@ -0,0 +1,161 @@
+        log.debug("get admin user info from " + provider + " : "  + user_info_resp + '\n' )
+        # example:
+        #
+        # {"login":"juniwang","id":8814383,"avatar_url":"https://avatars.githubusercontent.com/u/8814383?v=3","gravatar_id":"",
+        # "url":"https://api.github.com/users/juniwang","html_url":"https://github.com/juniwang",
+        # "followers_url":"https://api.github.com/users/juniwang/followers",
+        # "following_url":"https://api.github.com/users/juniwang/following{/other_user}",
+        # "gists_url":"https://api.github.com/users/juniwang/gists{/gist_id}",
+        # "starred_url":"https://api.github.com/users/juniwang/starred{/owner}{/repo}",
+        # "subscriptions_url":"https://api.github.com/users/juniwang/subscriptions",
+        # "organizations_url":"https://api.github.com/users/juniwang/orgs","repos_url":"https://api.github.com/users/juniwang/repos",
+        # "events_url":"https://api.github.com/users/juniwang/events{/privacy}",
+        # "received_events_url":"https://api.github.com/users/juniwang/received_events","type":"User","site_admin":false,
+        # "name":"Junbo Wang","company":"","blog":"","location":"Shanghai China",
+        # "email":"wangjunbo924@gmail.com","hireable":false,"bio":null,"public_repos":12,"public_gists":0,"followers":0,
+        # "following":1,"created_at":"2014-09-18T01:30:30Z","updated_at":"2014-11-25T09:00:37Z","private_gists":0,
+        # "total_private_repos":0,"owned_private_repos":0,"disk_usage":14179,"collaborators":0,
+        # "plan":{"name":"free","space":307200,"collaborators":0,"private_repos":0}}
+        #
+        user_info = json.loads(user_info_resp)
+        name = user_info["login"]
+        nickname = user_info["name"] if "name" in user_info else name
+        openid = str(user_info["id"])
+        avatar = user_info["avatar_url"]
+        # get user primary email
+        email_info_resp = get_remote(get_config('login/github/emails_info_url') + access_token)
+        log.debug("get email from github:" + email_info_resp + '\n')
+        # email_info include all user email provided by github
+        # email is user's primary email
+        email_info = json.loads(email_info_resp)
+        admin_with_token = user_manager.db_login(openid,
+                                                name=name,
+                                                nickname=nickname,
+                                                access_token=access_token,
+                                                email_info=email_info,
+                                                avatar_url=avatar)
+        # login flask
+        admin = admin_with_token["user"]
+        log.info("github user login successfully:" + repr(admin))
+        detail = user_manager.get_user_detail_info(user, hackathon_name=hackathon_name)
+        detail["token"] = admin_with_token["token"].token
+        return detail
+
+
+
+
+
+
+class LoginUser:
```

Comment:
remove the LoginUser and add “load_user(id)” in view.py
get AdminUser info from DB filter by id